docs/autoprocess.py

```diff
@@ -0,0 +1,412 @@
+            * If none of this options, then raise a ValueError
+
+        :param content: content of the directive
+        :type content: list
+        :param categories: set of all categories
+        :type categories: set
+        :param processes: dict of all github_url: processes pairs
+        :type processes:dict
+
+        :rtype: list
+        """
+        if not content:
+            return processes
+        else:
+            content = content[0]
+            # Make a filtered dict of all process that have name == content
+            querry_for_slug = dict(pair for pair in processes_dict.items() if pair[1]['slug'] == content)
+            if len(querry_for_slug.items()) == 1:
+                return querry_for_slug
+
+            elif content in categories:
+                if content == 'uncategorized':
+                    filtered_dict = dict(
+                        pair for pair in processes_dict.items() if 'category' not in pair[1])
+                    return filtered_dict
+                else:
+                    filtered_dict = dict(
+                        pair for pair in processes_dict.items()
+                        if 'category' in pair[1] and pair[1]['category'] == content)
+                    return filtered_dict
+
+            else:
+                raise ValueError("Invalid argument given to autoprocess directive.")
+
+    def get_all_processes(self, process_dir, github_base_url):
+        """
+        Return a dictionary of all processes in `process_dir`
+
+        Dictionary key is an URL's to process source code on resolwe-bio
+        GitHub page and value is process as it is loaded from yaml file.
+
+        :param process_dir: Parent directory of all process files
+        :type process_dir: path/str
+        :param github_base_url: Base of the url to resolwe-bio on GitHub
+        :type github_base_url: url/string
+
+        :rtype: dict
+        """
+        all_process_files = []
+        process_file_extensions = ['*.yaml', '*.yml']
+        for root, _, filenames in os.walk(process_dir):
+            for extension in process_file_extensions:
+                for filename in fnmatch.filter(filenames, extension):
+                    all_process_files.append(os.path.join(root, filename))
+
+        def read_yaml_file(fname):
+            with open(fname) as file_:
+                return yaml.load(file_)
+
+        all_processes = {}
+        for yaml_file in all_process_files:
+            processes_in_file = read_yaml_file(yaml_file)
+            for process in processes_in_file:
+
+                # This section finds the line in file where the
+                # defintion of the process starts. (there are
+                # multiple process definition in some files).
+                i = 1
+                found = False
+                with open(yaml_file) as file_:
+                    for line in file_:
+                        if 'slug: ' + process['slug'] in line:
```

Comment:
Now, that I've found out, why this is used for, I think that solution is sufficient. Solution would  be to look for first line of yaml element and return it when you find right slug, but this is overkill for this use-case.